src/platform/packages/shared/kbn-es-query-server/src/filter/simplified_filter.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+/**
+ * Validation Schemas for Simplified Filter Interface
+ *
+ * These schemas are used for server validation of API requests and responses
+ * in * as Code APIs.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+// ====================================================================
+// CORE FILTER OPERATOR AND VALUE SCHEMAS
+// ====================================================================
+
+/**
+ * Schema for range values used in numeric and date filters
+ */
+export const rangeValueSchema = schema.object({
+  gte: schema.maybe(
+    schema.oneOf([schema.number(), schema.string()], {
+      meta: { description: 'Greater than or equal to' },
+    })
+  ),
+  lte: schema.maybe(
+    schema.oneOf([schema.number(), schema.string()], {
+      meta: { description: 'Less than or equal to' },
+    })
+  ),
+  gt: schema.maybe(
+    schema.oneOf([schema.number(), schema.string()], {
+      meta: { description: 'Greater than' },
+    })
+  ),
+  lt: schema.maybe(
+    schema.oneOf([schema.number(), schema.string()], {
+      meta: { description: 'Less than' },
+    })
+  ),
+});
+
+/**
+ * Schema for all possible filter values
+ * Supports single values, arrays, and range objects
+ */
+export const filterValueSchema = schema.oneOf(
+  [
+    schema.string(),
+    schema.number(),
+    schema.boolean(),
+    schema.arrayOf(schema.oneOf([schema.string(), schema.number(), schema.boolean()])),
```

Comment:
Is it intentional that we can mix types in the array, and is it something filters actually support?